### PR TITLE
gtimelog: fix and update to latest master

### PR DESCRIPTION
**Pull request by nazarewk**
it was not working before, exiting with:
```
Typelib files for Gtk-3.0 are not available.

If you're on Ubuntu or another Debian-like distribution, please install
them with

    sudo apt install gir1.2-gtk-3.0
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nazarewk** [comment]
developed and confirmed working through https://github.com/nazarewk-iac/nix-configs/commit/3e2ea67f

**Comment by exarkun** [comment]
This is broken in 22.11 and is blocking me from upgrading.  @jonringer @FRidh Is there a change this could be merged? (Or at least reviewed)

**Comment by exarkun** [comment]
Ah darn.  Now I see that ofborg has pointed out an evaluation failure :/  The new expression tries to use `pythonPackages` but it is not defined...

**Comment by nazarewk** [comment]
ofborg should be happy now

File: pkgs/development/python-modules/gtimelog/default.nix [hunk 1]
```diff
@@ -1,27 +1,43 @@
...
+, harfbuzz
+, libsecret
+, libsoup
+, pango
 }:
-
 buildPythonPackage rec {
   pname = "gtimelog";
-  version = "unstable-2020-05-16";
+  version = "git-2022-10-28";
```
Comment (oxzi) [inline pkgs/development/python-modules/gtimelog/default.nix:19]: By [convention](https://nixos.org/manual/nixpkgs/stable/#sec-package-naming), it's `unstable-` for non tagged releases. But it's fine for me.

**Review by oxzi — APPROVED** [review]
Thanks for this PR. I have successfully built and ran the resulting binary.

**Comment by nazarewk** [comment]
@oxzi updated the version string

**Review by oxzi — APPROVED** [review]
LGTM.

Thanks for your PR and please excuse the delay on my response.

File: pkgs/development/python-modules/gtimelog/default.nix [hunk 2]
```diff
@@ -31,6 +47,26 @@ buildPythonPackage rec {
...
+        atk
+      ])}"
+    "--set GIO_MODULE_DIR ${lib.makeSearchPathOutput "out"
+      "lib/gio/modules" ([
+        glib-networking
+      ])}"
+  ];
+
   preFixup = ''
     wrapProgram $out/bin/gtimelog \
```
Comment (jtojnar) [inline pkgs/development/python-modules/gtimelog/default.nix:71]: The proper way is to add `wrapGappsHook` to `nativeBuildInputs`. Ideally with this in `preFixup` to avoid double wrapping: https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-common-issues-double-wrapped

File: pkgs/development/python-modules/gtimelog/default.nix [hunk 3]
```diff
@@ -42,18 +78,16 @@ buildPythonPackage rec {
...
       It's main goal is to be as unintrusive as possible.
-
       To run gtimelog successfully on a system that does not have full GNOME 3
       installed, the following NixOS options should be set:
       - programs.dconf.enable = true;
       - services.gnome.gnome-keyring.enable = true;
-
       In addition, the following packages should be added to the environment:
       - gnome.adwaita-icon-theme
       - gnome.dconf
```
Comment (jtojnar) [inline pkgs/development/python-modules/gtimelog/default.nix:87]: `gnome.dconf` has long been renamed to `dconf`. But `programs.dconf.enable = true;` is now sufficient.

File: pkgs/development/python-modules/gtimelog/default.nix [hunk 4]
```diff
@@ -42,18 +78,16 @@ buildPythonPackage rec {
     longDescription = ''
       GTimeLog is a small time tracking application for GNOME.
       It's main goal is to be as unintrusive as possible.
-
       To run gtimelog successfully on a system that does not have full GNOME 3
```
Comment (jtojnar) [inline pkgs/development/python-modules/gtimelog/default.nix:81]: GNOME 3 is not a thing either.

File: pkgs/development/python-modules/gtimelog/default.nix [hunk 5]
```diff
@@ -1,27 +1,43 @@
...
+, glib-networking
+, gobject-introspection
+, gtk3
+, harfbuzz
+, libsecret
+, libsoup
+, pango
 }:
-
 buildPythonPackage rec {
```
Comment (jtojnar) [inline pkgs/development/python-modules/gtimelog/default.nix:17]: Should not this be `buildPythonApplication`?

File: pkgs/development/python-modules/gtimelog/default.nix [hunk 6]
```diff
@@ -1,27 +1,43 @@
...
+    glibcLocales
+    gobject-introspection
+    gtk3
+    libsoup
+    libsecret
   ];
 
-  propagatedBuildInputs = [
-    pygobject3 freezegun mock
+  propagatedBuildInputs = with python.pkgs; [
```
Comment (jtojnar) [inline pkgs/development/python-modules/gtimelog/default.nix:37]: Why this? Stuff in `pkgs/development/python-modules` should just accept Python packages as arguments.

Not that apps that are not meant to be used as Python libraries belong in this directory. But if the file is moved, it should be done in a separate commit from the changes, so that Git can still track the history.

**Comment by akgrant43** [comment]
The original issue as reported by @nazarewk still exists in NixOS 23.05, which this patch resolves, so it has my vote for merging.

(I'm not commenting on @jtojnar 's review as I don't know enough).

Thanks!

File: pkgs/development/python-modules/gtimelog/default.nix [hunk 7]
```diff
@@ -1,27 +1,43 @@
...
+    pygobject3
+    freezegun
+    mock
```
Comment (jtojnar) [inline pkgs/development/python-modules/gtimelog/default.nix:40]: These two are for testing so should not be needed.

File: pkgs/development/python-modules/gtimelog/default.nix [hunk 8]
```diff
@@ -1,27 +1,43 @@
...
-    sha256 = "0qv2kv7vc3qqlzxsisgg31cmrkkqgnmxspbj10c5fhdmwzzwi0i9";
+    rev = "73d2228cea7e500858075a139fd321b1ccbae0d1";
+    sha256 = "sha256-g6gU95uvBhBbB4b1McE2LuVPDCjdmmhBX3lVyCY96zA=";
   };
 
   nativeBuildInputs = [ makeWrapper ];
   buildInputs = [
-    glibcLocales gobject-introspection gtk3 libsoup libsecret
+    glibcLocales
+    gobject-introspection
```
Comment (jtojnar) [inline pkgs/development/python-modules/gtimelog/default.nix:31]: This should go to `nativeBuildInputs`.

File: pkgs/development/python-modules/gtimelog/default.nix [hunk 9]
```diff
@@ -1,27 +1,43 @@
...
+, glibcLocales
+, glib-networking
+, gobject-introspection
+, gtk3
+, harfbuzz
+, libsecret
+, libsoup
+, pango
 }:
-
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gtimelog/default.nix:6]:
```suggestion

```

File: pkgs/development/python-modules/gtimelog/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/gtimelog/default.nix:17]: No, for python-packages that is correct

File: pkgs/development/python-modules/gtimelog/default.nix [hunk 10]
```diff
@@ -31,6 +47,26 @@ buildPythonPackage rec {
...
+        pango
+        harfbuzz
+        gdk-pixbuf
+        atk
+      ])}"
+    "--set GIO_MODULE_DIR ${lib.makeSearchPathOutput "out"
+      "lib/gio/modules" ([
+        glib-networking
+      ])}"
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/gtimelog/default.nix:68]: When moving gobject to nativeBuildInputs this should be done automatically

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/gtimelog/default.nix (hunk 5 above)
Comment (jtojnar) [reply pkgs/development/python-modules/gtimelog/default.nix:17]: This is not a Python package as far as I can tell, though. It might have been missed when we moved apps out of `python-modules/` directory.

**Comment by nazarewk** [comment]
please continue in https://github.com/NixOS/nixpkgs/pull/259395
